internal/pkg/fleetapi/audit_unenroll_cmd_test.go

```diff
@@ -0,0 +1,51 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package fleetapi
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/elastic/elastic-agent/internal/pkg/fleetapi/client"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_AuditUnenrollCmd_Execute(t *testing.T) {
+	const withAPIKey = "secret"
+	agentInfo := &agentinfo{}
+
+	t.Run("test audit/unenroll roundtrip", withServerWithAuthClient(
+		func(t *testing.T) *http.ServeMux {
+			mux := http.NewServeMux()
+			path := fmt.Sprintf(auditUnenrollPath, agentInfo.AgentID())
+			mux.HandleFunc(path, authHandler(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusOK)
```

Comment:
Why do we write a 200 OK header before validating the request body with the `require`s ?
maybe this handler can verify the request with `assert`s and return a 200 if everything is ok and something else (maybe a 400) if something is not what we expect ?

Reply:
this is a dummy handler; i'll add the validation but i don't expect that that the `fleetapi` level we care about the response (as the uninstall function controls retries)